src/Array.ts

```diff
@@ -182,6 +189,12 @@ export const findIndex = <A>(predicate: Predicate<A>) => (as: Array<A>): Option<
   return option.none
 }
 
+/** Find the first element which satisfies a predicate function */
+export const findFirst = <A>(predicate: Predicate<A>) => (as: Array<A>): Option<A> => head(filter(predicate)(as))
+
+/** Find the last element which satisfies a predicate function */
+export const findLast = <A>(predicate: Predicate<A>) => (as: Array<A>): Option<A> => last(filter(predicate)(as))
```

Comment:
`findLast` can be replaced with more efficient reversed `for`-cycle also wrapped with `fromNullable`